### PR TITLE
add table, user models to common

### PR DESCRIPTION
**Pull request by friendtocephalopods**
### Summary of Changes

Add table and user models from metadata builder to the common repo. Increase version number.

### Tests

None, just adding models.

### Documentation

None.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

**Comment by alran** [comment]
hm it seems like the test coverage check here isn't super useful. I'd be in favor of ignoring model files!

**Comment by friendtocephalopods** [comment]
Yep, that's the plan! I was briefly distracted by why tests weren't succeeding locally when they passed on CI (setup.py has flask but requirements does not). Also added mypy and flake8 for good measure.

**Comment by alran** [comment]
lgtm!

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "venv/bin/python3"
```
Comment (feng-tao) [inline .vscode/settings.json]: is this file needed?

Comment (friendtocephalopods) [reply .vscode/settings.json]: ack, no, the vscode ignore hadn't got in yet...I'll remove
